posydon/popsyn/star_formation_history.py

```diff
@@ -35,6 +36,596 @@
+        -------
+        float or array-like
+            The cosmic star formation rate density at the given redshift(s).
+        '''
+        SFR_interp = interp1d(self.redshifts, np.sum(self.SFR_data*self.dFOH, axis=1))
+        return SFR_interp(z)
+    
+class Fujimoto24(MadauBase):
+    '''The Fujimoto et al. (2024) star formation history model.
+    mean metallicity evolution of Madau & Fragos (2017).
+    
+    Fujimoto et al. (2024), ApJ SS, 275, 2, 36, 59
+    https://ui.adsabs.harvard.edu/abs/2024ApJS..275...36F/abstract
+    '''
+    def __init__(self, MODEL):
+        super().__init__(MODEL)
+        # Parameters for Fujimoto+24 CSFRD
+        self.CSFRD_params = {
+            "a": 0.010,
+            "b": 2.8,
+            "c": 3.3,
+            "d": 6.6,
+        }   
+    
+class Zalava21(MadauBase):
+    
+    def __init__(self, MODEL):
+        '''Initialise the Zalava+21 model
+        
+        Requires the following parameters:
+        - sub_model : str
+            Either min or max
+        '''
+        if 'sub_model' not in MODEL:
+            raise ValueError("Sub-model not given!")
+        
+        super().__init__(MODEL)
+        self._load_zalava_data()
+        
+    def _load_zalava_data(self):
+        '''Load the data from the Zalava+21 models
+        Transforms the data to the format used in the classes.
+        
+        '''  
+        data_file = os.path.join(PATH_TO_POSYDON_DATA, "SFR/Zalava+21.txt")
+        tmp_data = pd.read_csv(data_file, names=['redshift', 'SFRD_min', 'SFRD_max'], skiprows=1, sep='\s+')
+        self.redshifts = tmp_data['redshift'].values
+        if self.sub_model == 'min':
+            self.SFR_data = tmp_data['SFRD_min'].values
+        elif self.sub_model == 'max':
+            self.SFR_data = tmp_data['SFRD_max'].values
+        else:
+            raise ValueError("Invalid sub-model!")
+
+    # overwrite the CSFRD method of MadauBase        
+    def CSFRD(self, z):
+        SFR_interp = interp1d(self.redshifts, self.SFR_data)
+        return SFR_interp(z)
+        
+    
+def get_SFH_model(MODEL):
+    '''Return the appropriate SFH model based on the given parameters
+    
+    Parameters
+    ----------
+    MODEL : dict
+        Model parameters.
+    
+    Returns
+    -------
+    SFHBase
+        The SFH model instance.
+    '''
+    if MODEL["SFR"] == "Madau+Fragos17":
+        return MadauFragos17(MODEL)
+    elif MODEL['SFR'] == "Madau+Dickinson14":
+        return MadauDickinson14(MODEL)
+    elif MODEL['SFR'] == 'Fujimoto+24':
+        return Fujimoto24(MODEL)
+    elif MODEL['SFR'] == "Neijssel+19":
+        return Neijssel19(MODEL)
+    elif MODEL['SFR'] == "IllustrisTNG":
+        return IllustrisTNG(MODEL)
+    elif MODEL['SFR'] == "Chruslinska+21":
+        return Chruslinska21(MODEL)
+    elif MODEL['SFR'] == "Zalava+21":
+        return Zalava21(MODEL)
+    else:
+        raise ValueError("Invalid SFR!")
+
+def SFR_per_Z_at_z(z, met_bins, MODEL):
```

Comment:
```suggestion
def SFR_per_met_at_z(z, met_bins, MODEL):
```

I like the conciseness of the name, but `per_met_at_z` might be better for clarity of meaning in this case. What do you think?

Reply:
changed.